streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -385,7 +385,7 @@ public synchronized final InternalTopologyBuilder rewriteTopology(final StreamsC
         setApplicationId(config.getString(StreamsConfig.APPLICATION_ID_CONFIG));
 
         // maybe strip out caching layers
-        if (config.getLong(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG) == 0L) {
+        if (config.getLong(StreamsConfig.STATESTORE_CACHE_MAX_BYTES_CONFIG) == 0L) {
```

Comment:
Ah I wasn't trying to suggest we shouldn't start using the new config name everywhere, we absolutely should (even if it does explode the PR quite a bit -- I'll live). My point here was that we need to check the value of both of these configs, since we don't know which (if any) the user actually set.

Which, by the way,  you can check via `config.originals()` -- this returns the set of configs the user passed in, otherwise it's tricky to figure out since eg `config.getLong` would return the config default value if not set